deploy/Jenkinsfile

```diff
@@ -0,0 +1,227 @@
+pipeline {
+    agent {
+        label 'bb5'
+    }
+    stages {
+
+        stage('build AoS'){
+            steps {
+                sh 'sh deploy/install.sh AoS'
+            }
+        }
+
+        stage('build SoA'){
+            steps {
+                sh 'sh deploy/install.sh SoA'
+            }
+        }
+
+        stage('ringtest'){
+            parallel{
+                stage('AoS'){
+                    steps{
+                        sh 'sh deploy/ringtest.sh AoS'
+                    }
+                }
+                stage('SoA'){
+                    steps{
+                        sh 'sh deploy/ringtest.sh SoA'
+                    }
+                }
+            }
+        }
+
+        stage('checkout testcorenrn'){
+            steps{
+                sh 'cd $WORKSPACE'
+                sh 'rm -rf testcorenrn'
+                checkout([
+                    $class: 'GitSCM',
+                    branches: [[name: 'master']],
+                    doGenerateSubmoduleConfigurations: false,
+                    extensions: [[$class: 'RelativeTargetDirectory', relativeTargetDir: 'testcorenrn']],
+                    submoduleCfg: [],
+                    userRemoteConfigs: [[url: 'https://github.com/pramodk/testcorenrn.git']]
+                ])
+            }
+        }
+
+        stage('nrnivmodl'){
+            parallel{
+                stage('nrnivmodl'){
+                    steps{
+                        sh 'sh deploy/nrnivmodl.sh'
+                    }
+                }
+                stage('nrnivmodl-core AoS'){
+                    steps{
+                        sh 'sh deploy/nrnivmodl-core.sh AoS'
+                    }
+                }
+                stage('nrnivmodl-core SoA'){
+                    steps{
+                        sh 'sh deploy/nrnivmodl-core.sh SoA'
+                    }
+                }
+            }
+        }
+
+        stage('testcorenrn'){
```

Comment:
These stages can be created in a for loop. Something like:
```
script {
    def cmds = [:]
    for (vspec in VERSIONS_SPEC) {
        for (testname in TESTS) {
            def taskname = "${vspec.key}-${testname}"
            cmds[taskname] = """
                source .jenkins/testutils.sh
                run_test ${testname} ${vspec.value}
                """.toString()
        }
    }
    parallel cmds.collectEntries{
        key, cmd -> return [(key): { stage(key){sh(cmd)} } ] }
}
```

Reply:
After a lot of tries and research I could not find a way to construct with a `for-loop` the same pipeline design with running in parallel a branch with more than one stages, so I suppose that I will leave this part of the code like that. :disappointed: